llm/llama.cpp/generate_linux.go

```diff
@@ -16,7 +16,4 @@ package llm
 //go:generate cmake -S gguf -B gguf/build/cpu -DLLAMA_K_QUANTS=on
 //go:generate cmake --build gguf/build/cpu --target server --config Release
 
-//go:generate cmake -S ggml -B ggml/build/cuda -DLLAMA_CUBLAS=on -DLLAMA_ACCELERATE=on -DLLAMA_K_QUANTS=on
-//go:generate cmake --build ggml/build/cuda --target server --config Release
-//go:generate cmake -S gguf -B gguf/build/cuda -DLLAMA_CUBLAS=on -DLLAMA_ACCELERATE=on -DLLAMA_K_QUANTS=on
-//go:generate cmake --build gguf/build/cuda --target server --config Release
+//go:generate sh build_linux_cuda.sh
```

Comment:
Could we add `go:generate` commands for rocm (vs a script?) We'll probably need a way to selectively build rocm or cuda runners (and the "release" builds will include all runners)

Reply:
(also, thanks for opening this PR! 🤩 )

Reply:
Yeah, that was my initial version, to actually have a separate "rocm" binary instead of cuda. Is that preferable?

Reply:
One difficult issue (why I originally had a go "script" here) is that go generate doesn't appear to support conditional building :cry: or support tags...

Reply:
> Yeah, that was my initial version, to actually have a separate "rocm" binary instead of cuda. Is that preferable?

Oh! Sorry. Yes for sure, AFAIK, llama.cpp can be built with support for both cuda and rocm at the same time.

> One difficult issue (why I originally had a go "script" here) is that go generate doesn't appear to support conditional building 😢 or support tags...

Ah I just saw that too – shoot. Ok, a script will work okay then. Some thoughts:
1. Call it `generate_linux.sh`
2. Have simple env vars `GENERATE_CUDA` and `GENERATE_ROCM` and the script will generate respective runners based on what is set (by default it can generate cpu only). For releases we will build it with all runners.

Reply:
This might be possible with build tags, it seems like they may work with go generate, but still experimenting. That also allows swapping the CheckVRAM implementations more elegantly, if it works. I'll create a new pull with a better description if I get this working.

Reply:
PTAL #667 667